### PR TITLE
Changing 'Back to' link behavior

### PR DESCRIPTION
**Pull request by webinista**
Making 'Back to' link return the veteran to the folder of the current message thread.
- Closes department-of-veterans-affairs/messaging-fe#74
- Removing paths.DRAFTS_URL in favor of a systemFolders config option.
- Updating reference to paths.DRAFTS_URL in Compose.

### CODE REVIEW

File: src/js/messaging/components/compose/ModalConfirmDelete.jsx [hunk 1]
```diff
@@ -25,7 +25,7 @@ class ModalConfirmDelete extends React.Component {
           cssClass={this.props.cssClass}
           contents={modalContents}
           id={this.props.id}
-          onClose={this.props.onClose}
+          onClose={this.props.onDelete}
```
Comment (U-DON) [inline src/js/messaging/components/compose/ModalConfirmDelete.jsx]: Why should closing it trigger the delete?

Comment (webinista) [inline src/js/messaging/components/compose/ModalConfirmDelete.jsx]: Changed it in the wrong place. Thanks.

File: src/js/messaging/containers/Compose.jsx [hunk 2]
```diff
@@ -163,6 +166,7 @@ class Compose extends React.Component {
 
 const mapStateToProps = (state) => {
   return {
+    persistFolder: state.folders.data.currentItem.persistFolder,
```
Comment (U-DON) [inline src/js/messaging/containers/Compose.jsx]: Where is this being used? Was it intended for the redirect in `handleConfirmDelete`?

File: src/js/messaging/containers/Thread.jsx [hunk 3]
```diff
@@ -120,6 +120,7 @@ class Thread extends React.Component {
             currentMessageNumber={currentIndex + 1}
             moveToFolders={folders}
             folderMessageCount={folderMessageCount}
+            persistedFolder={this.props.persistFolder}
```
Comment (U-DON) [inline src/js/messaging/containers/Thread.jsx]: Add this to the propTypes of `ThreadHeader` too.

File: src/js/messaging/containers/Compose.jsx (hunk 2 above)
Comment (webinista) [inline src/js/messaging/containers/Compose.jsx]: It was. In department-of-veterans-affairs/messaging-fe#63, it's supposed to go return back to the folder the veteran clicked "Compose" from, but I want to push back on that a bit. That seems like a weird flow for the user.

**Comment by U-DON** [comment]
LGTM.

**Pull request merged**